### PR TITLE
set working folder to system.defaultworkingdirectory for node handler.

### PR DESCRIPTION
**Pull request by TingluoHuang**

### CODE REVIEW

File: src/Agent.Worker/Handlers/NodeHandler.cs [hunk 1]
```diff
@@ -43,7 +43,14 @@ public sealed class NodeHandler : Handler, INodeHandler
             string workingDirectory = Data.WorkingDirectory;
             if (string.IsNullOrEmpty(workingDirectory))
             {
-                workingDirectory = TaskDirectory;
+                if (!string.IsNullOrEmpty(ExecutionContext.Variables.System_DefaultWorkingDirectory))
+                {
```
Comment (bryanmacfarlane) [inline src/Agent.Worker/Handlers/NodeHandler.cs]: When would this not be set?  Couldn't this actually be an assert since it should never not be set?

Comment (TingluoHuang) [inline src/Agent.Worker/Handlers/NodeHandler.cs]: if you not sync source, the folder will not be there. (cleanup jobs)

Comment (bryanmacfarlane) [inline src/Agent.Worker/Handlers/NodeHandler.cs]: OK.  I still don't like TaskDirectory.  Is there another variable like the agent's root working directory?

Comment (TingluoHuang) [inline src/Agent.Worker/Handlers/NodeHandler.cs]: agent.workfolder?

Comment (bryanmacfarlane) [inline src/Agent.Worker/Handlers/NodeHandler.cs]: sounds good.

**Pull request merged**